src/Bitcoin/CashAddress.cpp

```diff
@@ -27,10 +24,10 @@ static const uint8_t p2shVersion = 0x08;
 static constexpr size_t maxHRPSize = 20;
 static constexpr size_t maxDataSize = 104;
 
-bool CashAddress::isValid(const std::string& string) {
+bool CashAddress::isValid(const std::string& cashHRP_, const std::string& string) {
```

Comment:
Just use `hrp`, no need to add `cash` prefix inside `CashAddress` class
```suggestion
bool CashAddress::isValid(const std::string& hrp, const std::string& string) {
```

Reply:
Second to that; also please avoid using underscore prefix&suffix in parameter names.

Reply:
OK. But I will need to find some way to disambiguate between the class member, the local variable (already called `hrp`) and the name of the parameter passed to the method.

Reply:
As a rule of thumb, class members can always by disamiguated by `this->` if needed.
I prefer underscore if needed in local variable, not in param name.
But this is minor issue, not really relevant.